app/models/pawn.rb

```diff
@@ -1,5 +1,76 @@
 class Pawn < Piece
 
+  def valid_move?(new_x, new_y)
+    on_board?(new_x, new_y) && passes_pawn_rules?(new_x, new_y)
+  end
+
+  def passes_pawn_rules?(x, y)
+    if two_squares_forward?(x, y)
+      return true
+    elsif one_square_forward?(x, y)
+      return true
+    elsif diagonal_move?(x, y)
+      return true
```

Comment:
@ilyakrasnov are we making different methods for i.e. en passant & pawn moving diagonal to capture pieces? If so, then diagonal_move can be taken out here?